File: 492.java
class Solution {
    public int[] constructRectangle(int area) {
        // time: O(logn), space: O(1)
        int w = (int) Math.sqrt(area);
        while (area % w != 0) {
            w--;
        }
        int l = area / w;
        return (new int[] {l, w});
    }
}

File: 492.py
from typing import List

class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        l = w = int(area ** 0.5)
        while True:
            if l * w == area:
                return [l, w]
            elif l * w < area:
                l += 1
            else:
                w -= 1